### PR TITLE
[MM-231] Sync the plugin with starter template

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Synced the plugin to include the below PRs:
- https://github.com/mattermost/mattermost-plugin-starter-template/pull/192
- https://github.com/mattermost/mattermost-plugin-starter-template/pull/193

#### Ticket Link

### CODE REVIEW

**Review by mickmister — APPROVED** [review]

**Comment by mickmister** [comment]
@ayusht2810 Can you please resolve the conflicts here? Thank you

**Review by mickmister — APPROVED** [review]

**Comment by mickmister** [comment]
@raghavaggarwal2308 Can you please take a look at this PR?

**Comment by mickmister** [comment]
Not sure if the lint one is running correctly though:

https://github.com/mattermost/mattermost-plugin-welcomebot/actions/runs/8344951987/job/22838648109?pr=123
image

**Review by raghavaggarwal2308 — APPROVED** [review]
LGTM, other than @mickmister comment above

**Comment by ayusht2810** [comment]
@mickmister Should we move the PR to QA review?

File: build/pluginctl/logs_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+				t.Logf("expected oldest: %v, got %v", tc.expectedOldest, oldest)
+				t.Fail()
+			}
+
+			compareSlice(t, tc.expectedLogs, logs)
+		})
+	}
+}
+
+func TestFilterLogEntries(t *testing.T) {
```
Comment (mickmister) [inline build/pluginctl/logs_test.go:98]: Does this contain the fix at https://github.com/mattermost/mattermost-plugin-starter-template/pull/197?

File: build/pluginctl/logs_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+			},
+			pluginID: "some.plugin.id",
+			expectedLogs: []string{
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53.091 +01:00"}`,
+			},
+			expectedErr: false,
+		},
+		"filter out old entries": {
+			logs: []string{
+				fmt.Sprintf(`{"message":"old2", "plugin_id": "some.plugin.id", "timestamp": "%s"}`, time.Now().Add(-2*time.Second).Format(timeStampFormat)),
```
Comment (mickmister) [inline build/pluginctl/logs_test.go:157]: I think we should apply the fix at https://github.com/mattermost/mattermost-plugin-starter-template/pull/197/files while this PR is open

**Review by mickmister — APPROVED** [review]

**Review by AayushChaudhary0001 — APPROVED** [review]
The above PR has been tested for all the basic slash commands and the functionality of plugin. LGTM. Approved

**Review by mickmister — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]

**Pull request merged**